### PR TITLE
First draft for workflow section

### PR DESCRIPTION
**Pull request by registerrier**
This is a draft for the analysis workflow.

We might decide on:
- how detailed low level analysis. Do we need a figure to show the whole data flow?
- Do we discuss here Npred and IRF factorization? Do we simply use a reference?

I have also added references to the relevant sub packages when presenting the workflow.

### CODE REVIEW

File: src/text/2-package.tex [hunk 1]
```diff
@@ -1,6 +1,105 @@
...
+% This might not be the bast place compared to introduction
+The data processing workflow in Gamma-ray astronomy is usually split in two sequences.
+The first one deals with the data processing from camera measurement, calibration, event
+reconstruction and selection to yield a list of reconstructed gamma-ray event candidates.
+This sequence, sometimes referred to as low-level analysis, is usually very specific to
+a given observation technique if not a given instrument.
+
+The other sequence, referred to as high-level analysis, deals with the extraction of physical
+quantities related to gamma-ray sources and the production of high-level products such as spectra,
+lightcurves and catalogs. The methods and tools applied here are more generic and are broadly
```
Comment (facero) [inline src/text/2-package.tex:13]: ==> such as sky maps, spectra, light curves

Comment (facero) [inline src/text/2-package.tex:13]: ==> such as sky maps, spectra, light curves

File: src/text/2-package.tex [hunk 2]
```diff
@@ -1,6 +1,105 @@
...
+shared across the field. They also frequently imply joint analysis of multi-instrument data.
+To extract physically relevant information, the measured data are usually compared to a
```
Comment (facero) [inline src/text/2-package.tex:15]: ==> To extract the physically relevant information of the observed gamma-ray sources, the measured data are compared to a sky model using statistical techniques such as maximum likelihood.

Comment (facero) [inline src/text/2-package.tex:15]: ==> To extract the physically relevant information of the observed gamma-ray sources, the measured data are compared to a sky model using statistical techniques such as maximum likelihood.

File: src/text/2-package.tex [hunk 3]
```diff
@@ -1,6 +1,105 @@
...
+  Gamma-ray instruments consider the probability density of the angular separation between true and reconstructed directions
+  $\delta p = p_{\rm true} - p$, i.e. $PSF(\delta p|p_{\rm true}, E_{\rm true})$.
+\item $E_{\rm disp}(E|p_{\rm true}, E_{\rm true})$ is the energy dispersion. It gives the probability to
+  reconstruct the photon at energy $E$ when the true energy is $E_{\rm true}$ and the true position :$p_{\rm true}$.
+  Gamma-ray instruments consider the probability density of the migration $\mu=\frac{E}{E_{\rm true}}$,
+  i.e. $E_{\rm disp}(\mu|p_{\rm true}, E_{\rm true})$.
+\end{itemize}
+
+
+Gamma-ray data at the Data Level 3 therefore consists in lists of gamma-like events and their
```
Comment (facero) [inline src/text/2-package.tex:54]: have the different data levels been introduced before ?

File: src/text/2-package.tex [hunk 4]
```diff
@@ -1,6 +1,105 @@
...
+corresponding instrument response functions (IRFs). The latter include the aforementioned
+effective area, point spread function (PSF), energy dispersion and residual hadronic background.
+The handling of DL3 data is performed by classes and methods
+in the gammapy.data  (see \ref{ssec:gammapy-data}) and the gammapy.irf
```
Comment (facero) [inline src/text/2-package.tex:58]: should the module names `gammapy.data` be in italic or something ? TBC with other paper blocks

Comment (facero) [inline src/text/2-package.tex:58]: should the module names `gammapy.data` be in italic or something ? TBC with other paper blocks

File: src/text/2-package.tex [hunk 5]
```diff
@@ -1,6 +1,105 @@
...
+(see \ref{ssec:gammapy-irf}) subpackages.
+
+The first step in the analysis is the selection and extraction of observations
+based of their meta data including information such as pointing direction, observation
+time and observation conditions.
+
+The next step of the analysis is the data reduction on a user-defined geometry.
+A typical geometry consists in a spectral representation with a measured
+energy axis, and in a spatial representation, either a coordinates system  with a projection
```
Comment (facero) [inline src/text/2-package.tex:67]: `a coordinates system` => `a coordinate system`

Comment (facero) [inline src/text/2-package.tex:67]: `a coordinates system` => `a coordinate system`

File: src/text/2-package.tex [hunk 6]
```diff
@@ -1,6 +1,105 @@
...
+effective area, point spread function (PSF), energy dispersion and residual hadronic background.
+The handling of DL3 data is performed by classes and methods
+in the gammapy.data  (see \ref{ssec:gammapy-data}) and the gammapy.irf
+(see \ref{ssec:gammapy-irf}) subpackages.
+
+The first step in the analysis is the selection and extraction of observations
+based of their meta data including information such as pointing direction, observation
+time and observation conditions.
+
+The next step of the analysis is the data reduction on a user-defined geometry.
```
Comment (facero) [inline src/text/2-package.tex:65]: I would move line 72 here and then define geometry.
Mixing a bit the text here is a suggestion:
`The next step of the analysis is the data reduction where all observation events and instrument responses are projected onto a user-defined geometry. 
a region on the sky (for regular spectral analysis)
For the 3-dimensional cube analysis the geometry consists of a sky coordinate system with a projection and a measured energy axis. For a regular 1-dimensional spectral analysis, the geometry consists of a spectral representation of the events extracted from a region on the sky.`

Comment (facero) [inline src/text/2-package.tex:65]: I would move line 72 here and then define geometry.
Mixing a bit the text here is a suggestion:
`The next step of the analysis is the data reduction where all observation events and instrument responses are projected onto a user-defined geometry. 
a region on the sky (for regular spectral analysis)
For the 3-dimensional cube analysis the geometry consists of a sky coordinate system with a projection and a measured energy axis. For a regular 1-dimensional spectral analysis, the geometry consists of a spectral representation of the events extracted from a region on the sky.`

File: src/text/2-package.tex [hunk 7]
```diff
@@ -1,6 +1,105 @@
...
+fit statistics. Different sub-classes support different analysis methods
+and fit statistics (e.g. Poisson statistics with known background or
+with OFF background measurements). The datasets are used to perform joint-likelihood
+fitting allowing to combine different measurements, e.g. from different observations
+but also from different instruments or event classes. They can also be used for binned
+simulation as well as event sampling to simulate DL3 events data.
+
+The next step is then typically to model and fit the datasets, either
+individually, or in a joint likelihood analysis. For this purpose Gammapy
+provides a uniform interface to multiple fitting backends. It also provides
```
Comment (facero) [inline src/text/2-package.tex:97]: `multiple fitting backends` provide examples ? (spicy-minize, iMinuit, emcee ?)

Comment (facero) [inline src/text/2-package.tex:97]: `multiple fitting backends` provide examples ? (spicy-minize, iMinuit, emcee ?)

File: src/text/2-package.tex [hunk 8]
```diff
@@ -1,6 +1,105 @@
...
+a variety of :ref:`built in models <model-gallery>`. This includes spectral,
+spatial and temporal model classes to describe the gamma-ray emission in the sky.
```
Comment (facero) [inline src/text/2-package.tex:99]: Suggestion:
Note that the spectral models can either be analytical (e.g. Power-laws) or from the radiation mechanisms of accelerated particle populations (e.g. inverse Compton or $\pi^{o}$ decay).

Comment (facero) [inline src/text/2-package.tex:99]: Suggestion:
Note that the spectral models can either be analytical (e.g. Power-laws) or from the radiation mechanisms of accelerated particle populations (e.g. inverse Compton or $\pi^{o}$ decay).

File: src/text/2-package.tex (hunk 1 above)
Comment (registerrier) [reply src/text/2-package.tex:13]: Done

File: src/text/2-package.tex (hunk 2 above)
Comment (registerrier) [reply src/text/2-package.tex:15]: OK

File: src/text/2-package.tex (hunk 4 above)
Comment (registerrier) [reply src/text/2-package.tex:58]: Yes, for now we haven't uniformized this.

Comment (adonath) [reply src/text/2-package.tex:58]: We currently don' have a uniform format for this. I can introducing one later and adapting all uses in the paper accordingly...

File: src/text/2-package.tex (hunk 7 above)
Comment (adonath) [reply src/text/2-package.tex:97]: I think there is no need for details here, the different fitting backends are mentioned later on the `gammapy.modeling` section.

File: src/text/2-package.tex [hunk 9]
```diff
@@ -1,6 +1,105 @@
 \section{Analysis Workflow Overview}
 \label{sec:analysis-workflow-overview}
-\todo{Regis and Fabio F.}
+
+% This might not be the bast place compared to introduction
+The data processing workflow in Gamma-ray astronomy is usually split in two sequences.
```
Comment (adonath) [inline src/text/2-package.tex:5]:
```suggestion
The data processing workflows  in Gamma-ray astronomy is usually split in two sequences.
```

**Comment by adonath** [comment]
@registerrier Do you have time to address the remaining comments by @facero?

**Comment by adonath** [comment]
@registerrier We can also go ahead an merge as is and then address remaining comments in the whole paper review...

**Comment by bkhelifi** [comment]
Dear @registerrier, I put here some minor comments for this draft:
- L5: Gamma-ray -> gamma-ray
- L9: if not a given instrument -> and even to a given instrument
- L73: 'target geometry'... in L65 the expression 'user-defined geometry' was used. To be unified
- L74: 'uncertainties background' -> 'uncertainties, background' ?

**Comment by adonath** [comment]
@registerrier Please address @bkhelifi comments.

**Review by adonath — APPROVED** [review]
Thanks @registerrier and @facero, I'll go ahead and merge this now. Further review will happen on the complete paper draft...

**Pull request merged**